packages/microsoft_kiota_serialization_json/lib/src/json_serialization_writer.dart

```diff
@@ -178,10 +178,15 @@ class JsonSerializationWriter implements SerializationWriter {
     }
     if (key?.isNotEmpty ?? false) {
       final objectContents = {..._contents};
+      dynamic value = objectContents;
```

Comment:
I'm not sure I understand why we need this here?
If we look at the [dotnet implementation](https://github.com/microsoft/kiota-dotnet/blob/273e23072904ded60e22a434013475975d80049a/src/serialization/json/JsonSerializationWriter.cs#L453) (or other languages), it doesn't have such a flattening logic.
To me, it seems that the generated serialization code might be incorrect and not doing the appropriate routing.

Can you please share the schema?

Reply:
Ah, see, the dotnet abstractions have an `IComposedTypeWrapper` and does serialization logic differently based on that. We don't have that yet, but we can align the Dart implementation with the C# impl if you want

Reply:
please go ahead!

Reply:
I checked both approaches and, at least to me, this seems to be the best way. Generation looks good, .NET just uses its types to do this kind of "unwrapping". In the end it results in the same JSON, so I think we're good now

Reply:
I'd really like the implementations/generated code to be as aligned as possible between the different languages to ease up troubleshooting and other aspects if you don't mind?

Reply:
We gain no benefit from this. Sure we can add the interface as a marker to those types, but it makes no difference in the serialization logic.